pkg/internal/apis/certmanager/install/pruning_test.go

```diff
@@ -17,22 +17,62 @@ limitations under the License.
 package install
 
 import (
-	"fmt"
+	"bytes"
+	"io/ioutil"
 	"testing"
 
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	apiextensionsinstall "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/install"
+	"k8s.io/apimachinery/pkg/runtime"
+	jsonserializer "k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/apimachinery/pkg/runtime/serializer/versioning"
+
 	crdfuzz "github.com/munnerz/crd-schema-fuzz"
 
 	"github.com/jetstack/cert-manager/pkg/api"
 	cmfuzzer "github.com/jetstack/cert-manager/pkg/internal/apis/certmanager/fuzzer"
 )
 
 func TestPruneTypes(t *testing.T) {
-	crdfuzz.SchemaFuzzTestForCRDWithPath(t, api.Scheme, crdPath("certificates"), cmfuzzer.Funcs)
-	crdfuzz.SchemaFuzzTestForCRDWithPath(t, api.Scheme, crdPath("certificaterequests"), cmfuzzer.Funcs)
-	crdfuzz.SchemaFuzzTestForCRDWithPath(t, api.Scheme, crdPath("issuers"), cmfuzzer.Funcs)
-	crdfuzz.SchemaFuzzTestForCRDWithPath(t, api.Scheme, crdPath("clusterissuers"), cmfuzzer.Funcs)
+	crdfuzz.SchemaFuzzTestForInternalCRD(t, api.Scheme, getCRD(t, "certificates.cert-manager.io"), cmfuzzer.Funcs)
+	crdfuzz.SchemaFuzzTestForInternalCRD(t, api.Scheme, getCRD(t, "certificaterequests.cert-manager.io"), cmfuzzer.Funcs)
+	crdfuzz.SchemaFuzzTestForInternalCRD(t, api.Scheme, getCRD(t, "issuers.cert-manager.io"), cmfuzzer.Funcs)
+	crdfuzz.SchemaFuzzTestForInternalCRD(t, api.Scheme, getCRD(t, "clusterissuers.cert-manager.io"), cmfuzzer.Funcs)
 }
 
-func crdPath(s string) string {
-	return fmt.Sprintf("../../../../../deploy/charts/cert-manager/crds/%s.yaml", s)
+func getCRD(t *testing.T, resourceName string) *apiextensions.CustomResourceDefinition {
```

Comment:
We may be better off to just create `pkg/internal/apis/testing` or something and define all of these different test cases there to avoid duplication of this sort of stuff 🤷‍♂